SpatialGDK/ExternalSchemaCodegen/Programs/Improbable.CodeGen.Unreal/Serialization.cs

```diff
@@ -0,0 +1,443 @@
+            var valueTypeName = Types.GetQualifiedTypeName(mapType.ValueType, bundle);
+            var mapName = $"{ fieldName }_map";
+            var kvPairName = "kvpair";
+            var builder = new StringBuilder();
+
+            builder.AppendLine($"{targetObjectName} = {Types.CollectionTypesToQualifiedTypes[Types.Collection.Map]}<{keyTypeName}, {valueTypeName}>();");
+            builder.AppendLine($"auto MapEntryCount = Schema_GetObjectCount({schemaObjectName}, {fieldId});");
+            builder.AppendLine($"for (uint32 i = 0; i < MapEntryCount; ++i)");
+            builder.AppendLine("{");
+            builder.AppendLine(Text.Indent(1, $@"Schema_Object* {kvPairName} = Schema_IndexObject({schemaObjectName}, {fieldId}, i);"));
+            builder.AppendLine(Text.Indent(1, $@"auto Key = {GetValueTypeDeserialization(mapType.KeyType, kvPairName, "SCHEMA_MAP_KEY_FIELD_ID")};"));
+            builder.AppendLine(Text.Indent(1, $@"auto Value = {GetValueTypeDeserialization(mapType.ValueType, kvPairName, "SCHEMA_MAP_VALUE_FIELD_ID")};"));
+            builder.AppendLine(Text.Indent(1, $@"{(targetIsOption ? $"(*{targetObjectName})" : targetObjectName)}[std::move(Key)] = std::move(Value);"));
+            builder.AppendLine("}");
+
+            if (wrapInBlock)
+            {
+                var blockWrapping = new StringBuilder();
+                blockWrapping.AppendLine("{");
+                blockWrapping.AppendLine(Text.Indent(1, builder.ToString()));
+                blockWrapping.AppendLine("}");
+                return blockWrapping.ToString().TrimEnd();
+            }
+
+            return builder.ToString();
+        }
+
+        private static string GetPrimitiveListDeserialization(PrimitiveType primitive, string schemaObjectName, string targetObjectName, string fieldName, string fieldId, Bundle bundle, bool targetIsOption = false)
+        {
+            if (primitive == PrimitiveType.Bytes)
+            {
+                return $"{targetObjectName} = ::improbable::utils::GetBytesList({schemaObjectName}, {fieldId});";
+            }
+            else if (primitive == PrimitiveType.String)
+            {
+                return $"{targetObjectName} = ::improbable::utils::GetStringList({schemaObjectName}, {fieldId});";
+            }
+
+            var listInnerType = Types.SchemaToCppTypes[primitive];
+            var builder = new StringBuilder();
+
+            builder.AppendLine($"uint32 {fieldName}Length = {GetPrimitiveCount(primitive, schemaObjectName, fieldId)};");
+            builder.AppendLine(Types.GetListInitialisation(targetObjectName, listInnerType, $"{fieldName}Length", targetIsOption));
+
+            targetObjectName = targetIsOption ? $"(*{targetObjectName})" : targetObjectName;
+            switch (primitive)
+            {
+                case PrimitiveType.Int32:
+                    builder.AppendLine($"Schema_GetInt32List({ schemaObjectName}, {fieldId}, {targetObjectName}.GetData());");
```

Comment:
Same comment as the other: interpolate the string version of the enum into place, collapse to one `AppendLine`